app/scripts/controllers/pod.js

```diff
@@ -87,6 +87,49 @@ angular.module('openshiftConsole')
       }
     };
 
+    /**
+     *  Will set the newTerm's isActive/isUsed to true, while deactivating the old's
+     */
+    $scope.onTerminalSelectChange = function(newTerm) {
+      // Deactivate all terminals
```

Comment:
Prob should update this comment, I don't think this is actually deactivating all the terminals every time you click a new option in the drop down (implies closing the terminal).